src/fieldAnalyser/fieldUtils.ts

```diff
@@ -36,3 +38,20 @@ export const createFields = async (
     await client.field.createFields(batch);
   }
 };
+
+export const createFieldsFromReport = async (
+  client: PlatformClient,
+  report: FieldAnalyserReport
+) => {
+  {
+    if (report.inconsistencies.size > 0) {
+      throw new FieldTypeInconsistencyError(report.inconsistencies);
+    }
+    await createFields(client, report.fields);
+  }
+};
+
+export const isFieldNameValid = (fieldName: string): boolean => {
+  const allowedChar = new RegExp('^[a-z]+[a-z0-9_]*$');
```

Comment:
Put the RegExp outside the scope of the function, that will reduce the amount of call to 'new Regexp'